client/core/core.go

```diff
@@ -1540,13 +1540,37 @@ func (c *Core) Logout() error {
 // initializeDEXConnections connects to the DEX servers in the conns map and
 // authenticates the connection. If registration is incomplete, reFee is run and
 // the connection will be authenticated once the `notifyfee` request is sent.
+// If an account is not found on the dex server upon dex authentication the
+// account is disabled, initialize() will be triggered which will result in the
+// user being prompted to register again.
 func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) []*DEXBrief {
 	// Connections will be attempted in parallel, so we'll need to protect the
 	// errorSet.
 	var wg sync.WaitGroup
 	c.connMtx.RLock()
-	defer c.connMtx.RUnlock()
 	results := make([]*DEXBrief, 0, len(c.conns))
+	var reconcileConnectionsWg sync.WaitGroup
+	reconcileConnectionsWg.Add(1)
+	disabledAccountHostChan := make(chan string)
+	disabledAccountHosts := make(map[string]bool)
+	// If an account has been disabled this will trigger core initialization.
+	go func() {
+		defer reconcileConnectionsWg.Done()
+		for disabledAccountHost := range disabledAccountHostChan {
+			disabledAccountHosts[disabledAccountHost] = true
+		}
+		if len(disabledAccountHosts) > 0 {
+			var enabledDEXConnections = make(map[string]*dexConnection)
+			for _, dc := range c.conns {
+				if _, ok := disabledAccountHosts[dc.acct.host]; !ok {
+					enabledDEXConnections[dc.acct.host] = dc
+				}
+			}
+			c.connMtx.Lock()
+			c.conns = enabledDEXConnections
+			c.connMtx.Unlock()
+		}
```

Comment:
I think can be a little more compact if you delete. Also maybe disconnect?
 i.e.:
```
			c.connMtx.Lock()
			for _, dc := range c.conns {
				if _, ok := disabledAccountHosts[dc.acct.host]; ok {
					dc.connMaster.Disconnect()
                                        delete(c.conns, dc.acct.host)
				}
			}
			c.connMtx.Unlock()
```

Reply:
I've made some changes here as suggested.